sonic-net/sonic-buildimage
# [build] Add the possibility to disable compilation of teamd and radv containers.

#### Why I did it
This optimization is needed for DPU SONiC. DPU SONiC runs a limited set of containers and teamd and radv containers are not part of them. Unlike the other containers, there was no possibility to disable teamd and radv containers compilation. 
To reduce DPU SONiC compilation time and reduce the image size this commit adds the possibility to disable their compilation.
#### How I did it
Two new configuration options are added to `rules/config` file:
- INCLUDE_TEAMD
- INCLUDE_ROUTER_ADVERTISER

By default to preserve the existing behavior both options are enabled. There are two ways to override them:
- To change option value to "n" in `rules/config` file.
- To override their value using `SONIC_OVERRIDE_BUILD_VARS` env variable:
`SONIC_OVERRIDE_BUILD_VARS="SONIC_INCLUDE_TEAMD=y SONIC_INCLUDE_ROUTER_ADVERTISER=n"`
#### How to verify it
The default behavior is preserved. To verify it compile the image without overriding new options. Install the image and verify that both teamd and radv containers are present and running.
To verify the new options override them with "n" value. Compile and install image. Verify that no docker containers are present. Verify that SWSS can start without errors.
#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog


#### Ensure to add label/tag for the feature raised. example - PR#2174 under sonic-utilities repo. where, Generic Config and Update feature has been labelled as GCU.

#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
